sweetest/src/test/java/com/mysugr/sweetest/LegacyPolymorphismTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.mysugr.sweetest
+
+import com.mysugr.sweetest.framework.base.BaseJUnitTest
+import com.mysugr.sweetest.framework.base.dependency
+import com.mysugr.sweetest.framework.configuration.moduleTestingConfiguration
+import com.mysugr.sweetest.framework.environment.TestEnvironment
+import org.junit.After
+import org.junit.Test
+
+class LegacyPolymorphismTest {
+
+    interface Animal
+
+    class Cat : Animal
+
+    @After
+    fun tearDown() {
+        TestEnvironment.fullReset()
+    }
+
+    @Test
+    fun `Consuming supertype gives subtype`() {
+
+        val config = moduleTestingConfiguration {
+            dependency realOnly of<Cat>()
```

Comment:
what would happen when we have another class with parent Animal configured here. 
If there is no test for it yet it would be nice to have one.

Reply:
added a test, like it?